keps/973-workload-priority/README.md

```diff
@@ -0,0 +1,349 @@
+  - [How workloads are created from Jobs](#how-workloads-are-created-from-jobs)
+    - [1. A job specifies both <code>workload's priority</code> and <code>pod's priority</code>](#1-a-job-specifies-both--and-)
+    - [2. A job specifies only <code>workload's priority</code>](#2-a-job-specifies-only-)
+    - [3. A job specifies only <code>pod's priority</code>](#3-a-job-specifies-only-)
+  - [How to expand Priority utility](#how-to-expand-priority-utility)
+  - [Where workload's Priority is used](#where-workloads-priority-is-used)
+  - [Not Implementing the workloadPriorityClass Controller](#not-implementing-the-workloadpriorityclass-controller)
+  - [What happens when a user changes the priority of <code>workloadPriorityClass</code>?](#what-happens-when-a-user-changes-the-priority-of-)
+  - [Future works](#future-works)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+In this proposal, a `WorkloadPriorityClass` is created.
+The `Workload` is able to utilize `WorkloadPriorityClass`.
+`WorkloadPriorityClass` is independent from pod's priority.
+The priority value is a part of the workload spec and is mutable.  
+In this document, the term "workload Priority" is used to refer
+to the priority utilized by Kueue controller for managing the queueing
+and preemption of workloads.  
+The term "pod Priority" is used to denote the priority utilized by the
+kube-scheduler for preempting pods and jobs.
+
+## Motivation
+
+Currently, some proposals are submitted for the kueue scheduling order.
+However, under the current implementation, the priority of the `Workload` is tied to the priority of the pod. Therefore, it's not possible to change only the priority of the `Workload`. We don't want to change the pod priority because pod's priority is tied to pod preemption. We need a mechanism where users can freely modify the priority of the `Workload` alone, not affecting pod priority.
+
+### Goals
+
+Implement `WorkloadPriorityClass`. `Workload` can utilize `WorkloadPriorityClass`.  
+CRDs like Job, MPIJob etc specify the `WorkloadPriorityClass` through labels.  
+Users can modify the priority of a `Workload` by changing `Workload`'s priority directly.
+
+### Non-Goals
+
+Using existing k8s `PriorityClass` for Workload's Priority is not recommended.  
+`WorkloadPriorityClass` doesn't implement all the features of the k8s `PriorityClass`
+because some fields on the k8s `PriorityClass` are not relevant to Kueue.
+
+## Proposal
+
+In this proposal, `WorkloadPriorityClass` is defined.
+The `Workload` is able to utilize this `WorkloadPriorityClass`.
+`WorkloadPriorityClass` is independent from pod's priority.
+The priority value will be part of the workload spec and be mutable.
```

Comment:
Does `priority value` mean including `priorityClassSource`?

Reply:
Thanks for your comment.
I don't mean `priority value` including `priorityClassSource` here, but `priorityClassSource` should be mutable when the workload isn't yet admitted by ClusterQueue.